src/useClickAway.ts

```diff
@@ -21,7 +25,7 @@ const useClickAway = (
         off(document, eventName, handler);
       }
     };
-  });
+  }, [events, ref]);
```

Comment:
`events` is an array, I though array cannot be a *dep*, maybe it should be like this:

```js
[...events, ref]
```

Reply:
profs?)

> React Hook useEffect has a spread element in its dependency array. This means we can't statically verify whether you've passed the correct dependencies. (react-hooks/exhaustive-deps)

Reply:
you're right

https://codesandbox.io/s/deps-array-kj8ov

Reply:
sorry, I'm trying to figure it out myself.

And you approach is totally wrong, with spread operator in `deps` react writes this warnings

```

Warning: The final argument passed to useEffect changed size between renders. The order and size of this array must remain constant.

Previous: [1, 2, 3, 1, 1, 1]
Incoming: [1, 2, 3]
    in App
```

Reply:
Could `useEffect` be changed to [`useDeepCompareEffect`](https://github.com/streamich/react-use/blob/master/docs/useDeepCompareEffect.md)?

Reply:
Why? `useEffect` works fine!
See codesandbox example

Reply:
I guess you are right. I was thinking if `events` is provided inline, it will be a new array every re-render.

```js
useClickAway(ref, fn, ['mousedown', 'touchstart'])
```